src/block-components/typography/style.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * External dependencies
+ */
+import {
+	getFontFamily, clampInheritedStyle, useStyles, getStyles,
+} from '~stackable/util'
+import { Style as StyleComponent } from '~stackable/components'
+
+const getStyleParams = ( options = {} ) => {
+	const {
+		selector,
+		inherit,
+		inheritMin,
+		inheritMax,
+		hoverSelector,
+	} = options
+
+	return [
+		{
+			selector,
+			styleRule: 'fontSize',
+			attrName: 'fontSize',
+			hasUnits: 'px',
+			responsive: 'all',
+			hover: 'all',
+			hoverSelector,
+			valueCallback: ( _value, getAttribute, device, state ) => {
+				let value = _value
+				const clampDesktopValue = inherit && clampInheritedStyle(
+					getAttribute( 'fontSize', 'desktop', state ),
+					{ min: inheritMin, max: inheritMax }
+				)
+
+				const clampTabletValue = clampInheritedStyle(
+					getAttribute( 'fontSize', 'tablet', state ),
+					{ min: inheritMin, max: inheritMax }
+				)
+				if ( device === 'tablet' ) {
+					if ( clampDesktopValue ) {
+						value = `${ clampDesktopValue }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					}
+					if ( getAttribute( 'fontSize', device, state ) !== '' ) {
+						value = `${ getAttribute( 'fontSize', device, state ) }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					}
+				}
+
+				if ( device === 'mobile' ) {
+					if ( clampDesktopValue ) {
+						value = `${ clampDesktopValue }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					}
+
+					if ( clampTabletValue ) {
+						value = `${ clampTabletValue }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					} else if ( clampDesktopValue || getAttribute( 'fontSize', device, state ) !== '' ) {
+						value = undefined
+					}
+
+					if ( getAttribute( 'fontSize', device, state ) !== '' ) {
+						value = `${ getAttribute( 'fontSize', device, state ) }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					}
+				}
+
```

Comment:
you can use `clampCallback` to generate the clamped value based on the inherited value